ui/src/stores/indexedDB.ts

```diff
@@ -70,13 +70,41 @@ export async function openDatabase(): Promise<IDBDatabase> {
     request.onsuccess = async (event) => {
       const db = (event.target as IDBOpenDBRequest).result;
       await ensureInitialState(db);
+      const general = await loadStateFromIndexedDB("general", db);
+      const shouldClearClipboard = await checkClipboardTimeout(general);
+
+      if (shouldClearClipboard && general?.clipboard) {
+        general.clipboard = {
+          nodes: [],
+          edges: [],
+          workflows: [],
+          copiedAt: null,
+        };
+        saveStateToIndexedDB({ clipboard: general.clipboard }, "general");
+      }
+
       resolve(db);
     };
 
     request.onerror = () => reject(request.error);
   });
 }
 
+async function checkClipboardTimeout(
```

Comment:
Having checkClipboardTimeout return a boolean is a little confusing. Renaming to something like isClipboardTimeoutExpired or something better would describe the function better